src/ORM/DataQuery.php

```diff
@@ -447,6 +447,40 @@ public function count()
         return $this->getFinalisedQuery()->count("DISTINCT {$quotedColumn}");
     }
 
+    /**
+     * Return whether this dataquery will have records. This will use `EXISTS` statements in SQL which are more
+     * performant - especially when used in combination with indexed columns (that you're filtering on)
+     *
+     * @return bool
+     */
+    public function exists()
+    {
+        // Grab a statement selecting "everything" - the engine shouldn't care what's being selected in an "EXISTS"
+        // statement anyway
+        $statement = $this->getFinalisedQuery();
+        $statement->setSelect('*');
```

Comment:
This MIGHT have to go inside the below `if (empty($having)) {` condition.

I suggest adding some more tests for grouping and having queries having ->exists() calls on them just to be sure. :)

Reply:
Good call. I really want to get this in to 4.4 so I'll follow up on this on Friday

Reply:
Ok I looked into this a bit. You're right that it "MIGHT". It's not part of the SQL standard to allow `HAVING` clauses using aliased columns. It doesn't work at all on PostgreSQL. MySQL lets you do it no problem though. I'll move it in for MySQL.